src/main/webapp/app/shared/import-list/import-table.component.html

```diff
@@ -0,0 +1,84 @@
+    <h5 class="m-0" [jhiTranslate]="columnBaseTranslationKey() + '.table.title'"></h5>
+    <div>
+        <div class="d-flex align-items-center">
+            @if (isLoading()) {
+                <fa-icon [icon]="faSpinner" animation="spin" />
+            }
+            <input
+                id="import-objects-search"
+                type="search"
+                class="form-control ms-2 rounded"
+                [placeholder]="'global.importTable.searchPlaceholder' | artemisTranslate"
+                aria-label="Search"
+                [(ngModel)]="searchTerm"
+                (ngModelChange)="search()"
+            />
+        </div>
+    </div>
+</div>
+<hr class="my-2" />
+<div class="import-table-container">
+    <table class="table align-middle table-striped">
+        <thead>
+            <tr>
+                <th class="col-1 clickable">
+                    <div class="d-flex align-items-center" (click)="setSortedColumn('ID')">
+                        <span>#</span>
+                        <ng-container
+                            *ngTemplateOutlet="sortIcon; context: { sortedColumn: 'ID', currentSortedColumn: sortedColumn(), sortingOrder: sortingOrder() }"
+                        ></ng-container>
+                    </div>
+                </th>
+                @for (column of columns(); track column) {
+                    <th class="col-4 clickable" (click)="setSortedColumn(column.name)">
+                        <div class="d-flex align-items-center">
+                            <span [jhiTranslate]="columnBaseTranslationKey() + '.table.columns.' + column.name"></span>
+                            <ng-container
+                                *ngTemplateOutlet="sortIcon; context: { sortedColumn: column.name, currentSortedColumn: sortedColumn(), sortingOrder: sortingOrder() }"
+                            ></ng-container>
+                        </div>
+                    </th>
+                }
+                <th></th>
+            </tr>
+        </thead>
+        <tbody>
+            @for (entity of resultsOnPage(); track entity.id) {
+                <tr class="flex-row">
+                    <td class="col-1">
+                        <span>{{ entity.id }}</span>
+                    </td>
+                    @for (column of columns(); track column) {
+                        <td class="text-break col-4">
+                            <ngb-highlight [result]="column.getProperty(entity)" [term]="searchTerm()" />
+                        </td>
+                    }
+                    <td class="col-1">
+                        <button class="btn btn-sm btn-outline-secondary" (click)="selectRow(entity)">Select</button>
+                    </td>
+                </tr>
+            } @empty {
+                <tr>
+                    <td [colSpan]="columns().length + 2" class="text-center text-secondary"><em jhiTranslate="global.generic.emptyList"></em></td>
```

Comment:
Why `+2`? Could you define the value in the template so this is more understandable what happens here?

Reply:
The +2 to is that the empty label spans over all columns including the ID and the action button column. Do you think I need a comment for that?

Reply:
I think a comment or defining the value with a descriptive value in the typescript would be good, yes